src/execute.rs

```diff
@@ -843,18 +843,19 @@ pub(crate) fn create_evm_verifier(
 }
 
 #[cfg(not(target_arch = "wasm32"))]
-pub(crate) fn create_evm_data_attestation_verifier(
+pub(crate) fn create_evm_data_attestation(
     vk_path: PathBuf,
     srs_path: PathBuf,
-    settings_path: PathBuf,
+    compiled_circuit_path: PathBuf,
     sol_code_path: PathBuf,
     abi_path: PathBuf,
     input: PathBuf,
 ) -> Result<(), Box<dyn Error>> {
-    use crate::graph::{DataSource, VarVisibility};
-    check_solc_requirement();
+    use crate::graph::{VarVisibility, DataSource};
 
-    let settings = GraphSettings::load(&settings_path)?;
+    check_solc_requirement();
+    let circuit = GraphCircuit::load(compiled_circuit_path)?;
+    let settings = circuit.core.settings.clone();
```

Comment:
how does this change the read in scales ? 

I think what we would want to do is: 

```rust
let input_scales = circuit.core.model.get_input_scales()
```